src/WebJobs.Extensions.DurableTask/LinuxAppServiceFileLogger.cs

```diff
@@ -173,9 +170,14 @@ private void RollFiles()
             // Rename current file to older file.
 
 #if !FUNCTIONS_V1
-            Syscall.rename(this.logFilePath, this.archiveFilePath);
+            rename(this.logFilePath, this.archiveFilePath);
 #endif
 
         }
+
+#if !FUNCTIONS_V1
+        [DllImport("libc", SetLastError = true)]
```

Comment:
Small nit: it looks like most of the resources I see that use P/Invoke with libc specifically callout "libc.so" instead of just "libc".

Reply:
I would agree with that, but `Mono` actually drops the `so` as seen here: https://github.com/mono/mono/blob/da11592cbea4269971f4b1f9624769a85cc10660/mcs/class/Mono.Posix/Mono.Posix/Syscall.cs#L323

Since we're trying to drop the `Mono` dependency (which we know does the right thing) I think there's good reason to implement this "Mono-style" :)

Reply:
Interestingly, the source I got this impression from is [Mono's documentation](https://www.mono-project.com/docs/advanced/pinvoke/). Source code always wins though!